File: kafka/src/main/java/com/us/kafka/KafkaProducerDemo.java
package com.us.kafka;

import org.apache.kafka.clients.producer.KafkaProducer;
import org.apache.kafka.clients.producer.Producer;
import org.apache.kafka.clients.producer.ProducerRecord;

import java.util.Properties;
import java.util.concurrent.TimeUnit;

/**
 * Created by yangyibo on 16/12/14.
 */
public class KafkaProducerDemo extends Thread {
    private static String topicName;

    private KafkaProducerDemo(String topicName) {
        super();
        this.topicName = topicName;

    }

    @Override
    public void run() {

        Producer<String, String> producer = getKafkaProducer();
        for (int i = 0; i < 10000; i++) {
            producer.send(new ProducerRecord<>(topicName, "message: 我是KafkaProducerDemo第" + i + "条信息 abel"));
            try {
                TimeUnit.SECONDS.sleep(3);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
        producer.close();
    }

    public KafkaProducer<String, String> getKafkaProducer() {
        Properties props = new Properties();
        props.put("bootstrap.servers", KafkaConfig.metadata_broker_list);
        props.put("acks", "all");
        props.put("retries", 3);
        props.put("batch.size", 16384);
        props.put("linger.ms", 1);
        props.put("buffer.memory", 33554432);
        props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
        props.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");
        return new KafkaProducer<>(props);
    }


    public static void creatKafkaProducer(String topicName) {
        new KafkaProducerDemo(topicName).start();// 使用kafka集群中创建好的主题 test
    }

}

File: spark/mySpark/src/main/scala/KafkaProducerDemo.scala
import java.util.HashMap

import org.apache.kafka.clients.producer.{KafkaProducer, ProducerConfig, ProducerRecord}
import org.apache.spark.{SparkConf, SparkContext}

import scala.collection.mutable.ArrayBuffer

/**
  * Created by yangyibo on 16/11/29.
  */
object KafkaProducerDemo {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setAppName("KafkaProducer")
    conf.setMaster("local[2]")
    val sc = new SparkContext(conf)
    val array = ArrayBuffer("one", "tow", "three")
    //      array.foreach(println)
    kafkaProducer(array)
    //      array.clear()
  }

  def kafkaProducer(args: ArrayBuffer[String]) {
    println("我有数据----------------------")
    if (args != null) {
      //      val brokers = "192.168.100.41:9092,192.168.100.42:9092,192.168.100.43:9092"
      val brokers = "192.168.100.71:9092"
      // Zookeeper connection properties
      val props = new HashMap[String, Object]()
      props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, brokers)
      props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
        "org.apache.kafka.common.serialization.StringSerializer")
      props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
        "org.apache.kafka.common.serialization.StringSerializer")
      val producer = new KafkaProducer[String, String](props)
      val topic = "abcd"
      // Send some messages
      while (true) {
        for (arg <- args) {
          println(arg + "----------我已经读去")
          val message = new ProducerRecord[String, String](topic, null, arg)
          println(producer.send(message).get().offset())

          Thread.sleep(1000)
        }
      }
      //        producer.close()
    }
  }
}
